dds/idl/typeobject_generator.h

```diff
@@ -54,19 +60,53 @@ class typeobject_generator : public dds_generator {
 
   void consider(Element& v, AST_Type* type, const std::string& anonymous_name);
   void strong_connect(AST_Type* type, const std::string& anonymous_name);
-  void generate_minimal_type_identifier(AST_Type* type, bool force_type_object);
+  void generate_type_identifier(AST_Type* type, bool force_type_object);
+
+  // Generate minimal and complete type objects and type identifiers for the corresponding types.
+  void generate_struct_type_identifier(AST_Type* type);
+  void generate_union_type_identifier(AST_Type* type);
+  void generate_enum_type_identifier(AST_Type* type);
+  void generate_array_type_identifier(AST_Type* type, bool force_type_object);
+  void generate_sequence_type_identifier(AST_Type* type, bool force_type_object);
+  void generate_alias_type_identifier(AST_Type* type);
+  void generate_primitive_type_identifier(AST_Type* type);
+  void update_maps(AST_Type* type,
+                   const OpenDDS::XTypes::TypeObject& minimal_to,
+                   const OpenDDS::XTypes::TypeObject& complete_to);
+  void set_builtin_member_annotations(AST_Decl* member,
+    OpenDDS::XTypes::Optional<OpenDDS::XTypes::AppliedBuiltinMemberAnnotations>& annotations);
+
   static bool name_sorter(const Element& x, const Element& y);
-  OpenDDS::XTypes::TypeObject get_minimal_type_object(AST_Type* type);
   OpenDDS::XTypes::TypeIdentifier get_minimal_type_identifier(AST_Type* type);
+  OpenDDS::XTypes::TypeIdentifier get_complete_type_identifier(AST_Type* type);
   bool generate(AST_Type* node, UTL_ScopedName* name);
 
-  typedef std::map<AST_Type*, OpenDDS::XTypes::TypeObject> MinimalTypeObjectMap;
-  MinimalTypeObjectMap minimal_type_object_map_;
-  typedef std::map<AST_Type*, OpenDDS::XTypes::TypeIdentifier> MinimalTypeIdentifierMap;
-  MinimalTypeIdentifierMap minimal_type_identifier_map_;
+  // Both fields must be constructed when an object is created.
+  typedef struct {
+    OpenDDS::XTypes::TypeObject minimal;
+    OpenDDS::XTypes::TypeObject complete;
+  } TypeObjectPair;
```

Comment:
```suggestion
  struct TypeObjectPair {
    OpenDDS::XTypes::TypeObject minimal;
    OpenDDS::XTypes::TypeObject complete;
  };
```

Declaring a struct like this is only useful in C. Same for the other structs below.